supafunc/errors.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+from typing_extensions import TypedDict
```

Comment:
TypedDict was added in Python 3.8 iirc and we declare min python version as 3.8 in our pyproject.toml here so it should be ok importing from just `typing` too (unless we want to support lower versions of python as well)